griptape/drivers/vector/opensearch_vector_store_driver.py

```diff
@@ -0,0 +1,184 @@
+
+        query_body = {
+            "size": 10000,
+            "query": {
+                "match_all": {}
+            }
+        }
+
+        # If a namespace is provided, adjust the query to match vectors in that namespace
+        if namespace:
+            query_body["query"] = {
+                "match": {
+                    "namespace": namespace
+                }
+            }
+
+        response = self.client.search(index=self.index_name, body=query_body)
+
+        entries = []
+        for hit in response["hits"]["hits"]:
+            vector_data = hit["_source"]
+            entry = BaseVectorStoreDriver.Entry(
+                id=hit["_id"],
+                vector=vector_data.get("vector"),
+                meta=vector_data.get("metadata"),
+                namespace=vector_data.get("namespace")
+            )
+            entries.append(entry)
+
+        return entries
+
+    def query(
+            self,
+            query_vector: list[float],
+            count: Optional[int] = None,
+            field_name: str = "vector",
+            namespace: Optional[str] = None,
+            include_vectors: bool = False,
+            include_metadata=True,
+            **kwargs
+    ) -> list[BaseVectorStoreDriver.QueryResult]:
+        count = count if count else BaseVectorStoreDriver.DEFAULT_QUERY_COUNT
+
+        # Base k-NN query
+        query_body = {
+            "size": count,
+            "query": {
+                "knn": {
+                    field_name: {
+                        "vector": query_vector,
+                        "k": count
+                    }
+                }
+            }
+        }
+
+        # If a namespace is provided, adjust the query to match vectors in that namespace
+        if namespace:
+            query_body["query"] = {
+                "bool": {
+                    "must": [
+                        {
+                            "match": {
+                                "namespace": namespace
+                            }
+                        },
+                        {
+                            "knn": {
+                                field_name: {
+                                    "vector": query_vector,
+                                    "k": count
+                                }
+                            }
+                        }
+                    ]
+                }
+            }
+
+        response = self.client.search(index=self.index_name, body=query_body)
+
+        return [
+            BaseVectorStoreDriver.QueryResult(
+                namespace=hit["_source"].get("namespace") if namespace else None,
+                score=hit["_score"],
+                vector=hit["_source"].get("vector") if include_vectors else None,
+                meta=hit["_source"].get("metadata") if include_metadata else None
+            )
+            for hit in response["hits"]["hits"]
+        ]
+
+    def initialize_index(self):
```

Comment:
This should be named `create_index`, so it's consistent with other drivers. Should also accept `namespace` and `dimensions`.